### PR TITLE
make DotnetVersion optional, unpin dotnetversion.  Added setup task t…

### PR DESCRIPTION
**Pull request by david-driscoll**
…o better restrict types.  Reacted to --version -> --info from dotnet

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -249,21 +269,31 @@ Task("BuildEnvironment")
...
     DownloadFile($"{buildPlan.DotNetInstallScriptURL}/{installScript}", scriptPath);
-    if (!IsRunningOnWindows())
+    string installArgs = "";
+
+    if (IsRunningOnWindows())
+    {
+        installArgs = $"{buildPlan.DotNetChannel}";
+        if (!String.IsNullOrEmpty(buildPlan.DotNetVersion))
+            installArgs = $"{installArgs} -version {buildPlan.DotNetVersion}";
+        if (!buildPlan.UseSystemDotNetPath)
```
Comment (david-driscoll) [inline build.cake]: I wish these were consistent between platforms... *cough* *cough*

**Comment by david-driscoll** [comment]
cc @troydai @DustinCampbell @hal-ler

**Comment by hal-ler** [comment]
You can also use "Latest" as the version string to unpin, without having to add an additional check in build.cake..

File: build.cake (hunk 1 above)
Comment (hal-ler) [inline build.cake]: Oh I just assumed that the guys followed standard *NIX argument naming convention where full argument names use "--". But it looks like they also recently added the Windows convention with "-version" and such.

**Comment by troydai** [comment]
:shipit:

**Pull request merged**